stripe/src/main/java/com/stripe/android/ApiOperation.kt

```diff
@@ -0,0 +1,32 @@
+package com.stripe.android
+
+import android.os.AsyncTask
+import com.stripe.android.exception.StripeException
+import org.json.JSONException
+
+internal abstract class ApiOperation<ResultType>(
+    private val callback: ApiResultCallback<ResultType>
+) : AsyncTask<Void, Void, ResultWrapper<ResultType>>() {
+
+    internal abstract fun getResult(): ResultType?
+
+    override fun doInBackground(vararg voids: Void): ResultWrapper<ResultType> {
+        return try {
+            ResultWrapper.create(getResult())
+        } catch (e: StripeException) {
```

Comment:
alternatively, when statement w/ a single catch based on the type of e

Reply:
I can do in followup